src/theories/Functor.jl

```diff
@@ -0,0 +1,23 @@
+export AbstractFunctor, AbstractMonoidalFunctor
+
+abstract type AbstractFunctor end
+
+function dom(F::AbstractFunctor)::Category end
+function codom(F::AbstractFunctor)::Category end
+
+function (F::AbstractFunctor)(f::HomExpr{:compose})
+  compose(map(x->F(x), f.args))
```

Comment:
I could be wrong, but I think we can make it even simpler by writing `map(F, f.args)`.